algorithm/cgal/FromPolygonSoup.h

```diff
@@ -5,55 +5,75 @@ int FromPolygonSoup(Geometry* geometry, emscripten::val fill, bool wrap_always,
+      CGAL::Polygon_mesh_processing::orient_polygon_soup(points, polygons);
 
-  CGAL::Polygon_mesh_processing::repair_polygon_soup(points, polygons);
-  CGAL::Polygon_mesh_processing::orient_polygon_soup(points, polygons);
-
-  int target = geometry->add(GEOMETRY_MESH);
-  Surface_mesh& mesh = geometry->mesh(target);
-  geometry->setIdentityTransform(target);
-
-  CGAL::Polygon_mesh_processing::polygon_soup_to_polygon_mesh(points, polygons,
-                                                              mesh);
-
-  if (wrap_always || CGAL::Polygon_mesh_processing::does_self_intersect(mesh)) {
-    // Use a wrapping pass to remove self-intersection.
-    CGAL::Cartesian_converter<Kernel, Epick_kernel> to_cartesian;
-    Epick_points points;
-    std::vector<std::vector<size_t>> faces;
-    wrap_add_mesh_epick(to_cartesian, mesh, points, faces);
-    double alpha;
-    double offset;
-    if (wrap_absolute_alpha == 0 && wrap_absolute_offset == 0) {
-      wrap_compute_alpha_and_offset(CGAL::Polygon_mesh_processing::bbox(mesh),
-                                    wrap_relative_alpha, wrap_relative_offset,
-                                    alpha, offset);
-    } else {
-      alpha = wrap_absolute_alpha;
-      offset = wrap_absolute_offset;
+      CGAL::Polygon_mesh_processing::polygon_soup_to_polygon_mesh(
+          points, polygons, mesh);
     }
-    mesh.clear();
-    wrap_epick(points, faces, alpha, offset, mesh);
-  }
 
-  if (corner_threshold > 0) {
-    // Enable simplification to reduce polygon count.
-    simplify(corner_threshold, mesh);
-  }
+    if (wrap_always ||
+        CGAL::Polygon_mesh_processing::does_self_intersect(mesh)) {
+      std::cout << "QQ/FromPolygonSoup/Wrap" << std::endl;
+      // Use a wrapping pass to remove self-intersection.
+      CGAL::Cartesian_converter<Kernel, Epick_kernel> to_cartesian;
+      Epick_points points;
+      std::vector<std::vector<size_t>> faces;
+      std::cout << "QQ/FromPolygonSoup/Wrap/AddMesh" << std::endl;
+      wrap_add_mesh_epick(to_cartesian, mesh, points, faces);
+      double alpha;
+      double offset;
+      if (wrap_absolute_alpha == 0 && wrap_absolute_offset == 0) {
+        std::cout << "QQ/FromPolygonSoup/Wrap/ComputeAlpha" << std::endl;
+        wrap_compute_alpha_and_offset(CGAL::Polygon_mesh_processing::bbox(mesh),
+                                      wrap_relative_alpha, wrap_relative_offset,
+                                      alpha, offset);
+      } else {
+        alpha = wrap_absolute_alpha;
+        offset = wrap_absolute_offset;
+      }
+      std::cout << "QQ/FromPolygonSoup/Wrap: alpha=" << alpha
+                << " offset=" << offset << std::endl;
+      std::cout << "QQ/FromPolygonSoup/Wrap/MeshClear" << std::endl;
+      mesh.clear();
+      std::cout << "QQ/FromPolygonSoup/Wrap/Wrap" << std::endl;
+      wrap_epick(points, faces, alpha, offset, mesh);
+    }
 
-  demesh(mesh);
+    if (corner_threshold > 0) {
```

Comment:
This is a nice way to do it. No need for a simplification flag and a corner threshold value